src/ledger.rs

```diff
@@ -77,6 +90,49 @@ impl FileLedger {
         Self::default()
     }
 
+    // --- Historical Root Management ---
```

Comment:
we would need a function to prune the `historical_roots` by just leaving N latest or maybe make it part of the record_current_root?
I'm not sure how to make it more elegant and easy for the indexer to use and not screw up.